lib/Parse/ParseSIL.cpp

```diff
@@ -1108,13 +1105,21 @@ bool SILParser::parseSILDottedPathWithoutPound(ValueDecl *&Decl,
                                    SmallVectorImpl<ValueDecl *> &values) {
   // Handle sil-dotted-path.
   Identifier Id;
-  SmallVector<Identifier, 4> FullName;
+  SmallVector<DeclBaseName, 4> FullName;
   SmallVector<SourceLoc, 4> Locs;
   do {
     Locs.push_back(P.Tok.getLoc());
-    if (parseSILIdentifier(Id, diag::expected_sil_constant))
-      return true;
-    FullName.push_back(Id);
+    switch (P.Tok.getKind()) {
+    case tok::kw_subscript:
+      P.consumeToken();
+      FullName.push_back(DeclBaseName::createSubscript());
```

Comment:
I don't know if it answers the question but maybe I'll just comment on what's going on here:

Previously, subscripts were backed by the identifier "subscript" so no distinction between properties named "subscript" and proper subscripts was necessary. Now we need distinguish between the two. As per https://github.com/apple/swift/pull/6848#issuecomment-275792066 I decided to represent a property named "subscript" as "\`subscript\`" in a SIL dotted path while plain "subscript" (without the backticks) still refers to a proper subscript.

So what I'm doing here is checking if a component in a SIL dotted path is the keyword subscript in order to translate it to a special DeclBaseName for subscripts. If "subscript" is spelled in backticks, the token kind will be "backtick", the default branch is taken and `parseSILIdentifier` will parse ``"`subscript`"`` as an identifier named "subscript", producing a dotted path that refers to a property named "subscript".